### PR TITLE
RLE part 2 (PR 1 of 2)

### PR DESCRIPTION
**Pull request by AaronM04**
Oof

### CODE REVIEW

File: src/universe.rs [hunk 1]
```diff
@@ -1446,79 +1451,163 @@ impl Universe {
...
-        let cut_idx = cmp::max(gen0_idx, gen1_idx);
-        {
-            let (lower, upper): (&mut [GenState], &mut [GenState]) = self.gen_states.split_at_mut(cut_idx);
-            if gen1_idx < cut_idx {
-                lower[gen1_idx].clear();
-                upper[gen0_idx - cut_idx].copy(&mut lower[gen1_idx]); // this is an |= operation, hence the clear before this
-            } else {
-                upper[gen1_idx - cut_idx].clear();
-                lower[gen0_idx].copy(&mut upper[gen1_idx - cut_idx]); // this is an |= operation, hence the clear before this
+        if diff.gen0 > 0 {
```
Comment (Manghi) [inline src/universe.rs:1547]: Could perhaps use `inline` functions to describe what this is doing for clean-up/readability.

**Review by Manghi — APPROVED** [review]
Nicely done. Really like the introduction of `ConwayError`. I think I will do something similar for Netwayste :D

File: src/universe.rs (hunk 1 above)
Comment (AaronM04) [reply src/universe.rs:1547]: Yeah, I'd like to move this to a function. I felt a little dirty writing it like this...

**Comment by AaronM04** [comment]
Thanks @Manghi !

**Pull request merged**